views/js/controller/Delivery/monitoring.js

```diff
@@ -472,16 +474,17 @@ define([
                 }
 
                 /**
-                 * Check if an action is available with respect to the provided state
+                 * Check if an action is available for a provided delivery with respect to the provided state
                  * @param {String} what
-                 * @param {Object} state
+                 * @param {Object} delivery
                  * @returns {Boolean}
                  */
-                function canDo(what, state) {
-                    var status;
-                    if (state && state.status) {
-                        status = _status.getStatusByCode(state.status);
-                        return status && status.can[what] === true;
+                function canDo(what, delivery) {
+                    if (delivery && delivery.state.status) {
+                        const status = _status.getStatusByCode(delivery.state.status);
+                        const canDo = _.isFunction(status.can[what]) ?status.can[what](delivery) : status.can[what];
```

Comment:
It would be great if you could apply prettier and rename this variable to make this part of code a cleaner.
I haven't checked if every action (callback) returns `true` in case of execution `status.can[what](delivery)`, so I have some doubts about this solution 🤔

Reply:
Hi @andrey-ponamarev , no, the previous version of this function and the current version does the same: check if a given action is available for a given state. What has changed is that until now available actions for each state was a fixed truly/falsy value, and now it can be a dynamic value managed by a function. To do so I had to make two changes on this function:

1- Instead of just receiving the delivery state as a param, now it receives the full delivery so it can be used if required to evaluate a permission (state is still accesible via `delivery.state`).
2- Need to check if the permission is managed by a function so instead of returning the function I execute it (line 485).